web/concrete/src/Application/Application.php

```diff
@@ -58,9 +58,9 @@ public function shutdown($options = array())
             }
 
             $logger = new Logger();
-            $r = Request::getInstance();
+            $r = $this['request']->getInstance();
```

Comment:
Well if $this['request'] is the same as `\Request::getInstance()` always, that's fine. The other issue I point out is that the Request class itself is in charge of creating and keeping track of the current instance. And that's true for other singleton classes we have as well.

One of the most painful is the User class constructor, I've been thinking about that a lot too..